src/test/java/com/znsio/sample/e2e/steps/AmazonSteps.java

```diff
@@ -0,0 +1,61 @@
+package com.znsio.sample.e2e.steps;
+
+import com.context.SessionContext;
+import com.context.TestExecutionContext;
+import com.znsio.e2e.entities.TEST_CONTEXT;
+import com.znsio.e2e.runner.Runner;
+import com.znsio.e2e.tools.Drivers;
+import com.znsio.sample.e2e.businessLayer.amazon.HomePageBL;
+import com.znsio.sample.e2e.businessLayer.amazon.ProductInformationBL;
+import com.znsio.sample.e2e.businessLayer.amazon.SearchBL;
+import com.znsio.sample.e2e.businessLayer.amazon.ShoppingCartBL;
+import com.znsio.sample.e2e.entities.SAMPLE_TEST_CONTEXT;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.log4j.Logger;
+
+public class AmazonSteps {
+
+    private static final Logger LOGGER = Logger.getLogger(AmazonSteps.class.getName());
+    private final TestExecutionContext context;
+    private final Drivers allDrivers;
+
+    public AmazonSteps(){
+        context = SessionContext.getTestExecutionContext(Thread.currentThread()
+                .getId());
+        LOGGER.info("context: " + context.getTestName());
+        allDrivers = (Drivers) context.getTestState(TEST_CONTEXT.ALL_DRIVERS);
+        LOGGER.info("allDrivers: " + (null == allDrivers));
+    }
+
+    @Given("^user is on amazon home page$")
+    public void userIsLoggedInWithValidCredentialsToAmazon() {
+        allDrivers.createDriverFor(SAMPLE_TEST_CONTEXT.ME, Runner.platform, context);
+        LOGGER.info("User is about to Launch Amazon Home Page");
+        new HomePageBL(SAMPLE_TEST_CONTEXT.ME, Runner.platform).launchHomePageAmazon();
+        LOGGER.info("User is On Amazon Home Page");
+    }
+
+
+    @When("user searches for {string}")
+    public void userSearchesFor(String productName) {
+        LOGGER.info("User searches for product: "+ productName);
+        new HomePageBL().userSearchesForProduct(productName);
+    }
+
+
+    @And("user adds the product to shopping cart")
+    public void userAddsTheProductToShoppingCart() {
+        new SearchBL().userClicksOnFirstProduct();
```

Comment:
can be used as chaining as the userClicksOnFirstProduct() Return the same that we needed for second